tensorflow/lite/core/subgraph.cc

```diff
@@ -169,7 +169,6 @@ Subgraph::~Subgraph() {
     TfLiteIntArrayFree(node.inputs);
     TfLiteIntArrayFree(node.outputs);
     TfLiteIntArrayFree(node.temporaries);
-    if (node.builtin_data) free(node.builtin_data);
```

Comment:
You've only used unique_ptr for the tests. However, this will break real-world invocation (see also `lite/model.cc` which supplies the malloc'ed builtin data buffers).

Reply:
@jdduke , we need to change this everywhere and make sure that the implementation does not  force the users of the interface to allocate memory only via malloc (as i think this is wrong at so many levels), the better method would be to use the smart pointer and give the responsibility of the allocating and freeing the memory on the caller of the interface.

          Since the change is little big i wanted your opinion before i start making the change, let me know what you feel, if you are OK i will make the changes.

Regards
Amit

Reply:
> not force the users of the interface to allocate memory only via malloc

Can you describe your use-case? The API for installing a node with builtin_data is not really part of the public API. All clients should be using `InterpreterBuilder` for creating the `Interpreter`, so this is really an implementation detail, and kernel usage of builtin_data is effectively read-only.

Reply:
@jdduke , i get your point, but why to use this inside also if other options are available.  Anyway it was great talking to you, hope for more such interactions in the future. I am closing the PR.

Regards
Amit